moped-editor/src/views/projects/projectView/ProjectComponents/EditAttributesModal.js

```diff
@@ -52,7 +52,7 @@ const EditComponentModal = ({
   const onSave = (formData) => {
     const isSavingSignalFeature = Boolean(formData.signal);
 
-    const { description, subcomponents } = formData;
+    const { description, subcomponents, phase, subphase, completionDate } = formData;
```

Comment:
I noticed that if you edit an existing component phase, clear the phase with the X in the phase form, and then save, it will remove the phase but the previous completion date is left behind in the database. It isn't harmful because the UI blanks out the next time around, and future phase edits will overwrite the completion date left behind. 

I think it would be good to clean out the old values in the DB here by adding a check if a phase is not null and then assigning the completion data value that is fed to the mutation and state updates below.

To reproduce:
1. Add a component phase, subphase, and completion date and save
2. Edit the component attributes and remove the phase only with the X icon and save.
3. Edit again and the phase, subphase, and completion date don't fill existing values which is good
4. If you look in the Hasura console, the completion date is sneakily still hanging around

Reply:
thanks for catching that! i'll try to set it so clear the completion date when you clear the phase